### PR TITLE
[PyTorch] fix(Batch Normalization): memory leaky

### PR DESCRIPTION
**Pull request by mzz2017**
The memory would not be freed and accumulate every epochs to explode finally.

X is associated with *moving_mean* and *moving_var* via gradient computation.

As the result, **X will not be freed** due to the resident of *self.moving_mean* and *self.moving_var* in memory.

Thus the returning values here should be modified to *moving_mean.data* and *moving_var.data* to remove the gradient property, which will not impact our model.

### CODE REVIEW

**Comment by mzz2017** [comment]
I just modified the code in pytorch version.
Please check and update it if there is the same problem in mxnet and tensorflow, thanks.

**Comment by mli** [comment]
Job d2l-en/PR-1342/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1342/

**Comment by mzz2017** [comment]
Thanks and sorry for duplication

**Comment by mli** [comment]
Job d2l-en/PR-1342/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1342/

**Comment by mli** [comment]
Job d2l-en/PR-1342/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1342/
